sycl/doc/extensions/proposed/sycl_ext_oneapi_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,404 @@
+for which `get_local_linear_id` return 0.
+|===
+
+
+To obtain the total number of clusters in the nd_range, and to obtain the 
+id of the cluster of the calling work-item, this extension proposes to add two
+new member functions the `nd_item` class, namely 
+`ext_oneapi_cuda_get_cluster_range` and `ext_oneapi_cuda_get_cluster_id`.
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|`range<Dim> nd_item::ext_oneapi_cuda_get_cluster_range() const`
+|Returns the total number of clusters across each dimension.
+
+|`id<Dim> nd_item::get_cluster_id() const`
+|Returns the id of the cluster along each dimension.
+|===
+
+
+== Accessing another work-group's local memory
+
+Work-group's within the cluster have the ability to access another work-group's 
+local memory. Typically addresses which reside in the local memory of a 
+work-group can only be addressed by the work-items of that work-group. 
+Therefore, to access another work-group's local memory, the address needs to be
+mapped such that the address in another work-group is addressable within the 
+calling work-item. Further, to access another another work-group's local memory,
+all the work-groups within the cluster must exist and the work-group's should
+not cease to exist before all the memory operations are completed. This can be 
+ensured by synchronizing all the work-items within the cluster before and after
+the local memory operations, using `group_barrier`.
+
+A member function of the `cluster_group` class; 
+`map_cluster_local_pointer` will perform the mapping and return a pointer 
+which can then be dereferenced by the calling work-item. 
+
+
+[%header,cols="10,5"]
+|===
+|Method
+|Description
+
+|T*  map_cluster_local_pointer(T* addr, size_t group_id)
+|Accepts the equivalent address to the memory location relative to the calling 
+work-item which is to be mapped from the local memory of the work-group, as 
+specified by the group_id, denoting the linear group_id within the cluster
+|===
+
+
+== Cluster Memory Fence Scope and Barrier
+
+Work-items in a work-group can atomically operate on the local memory addresses
+of another work-group which have mapped as described above. To facilitate this,
+this proposal introduces a new memory scope, 
+`sycl_ext_oneapi_experimental_cuda_cluster_fence` which can be used with the
+existing `atomic_ref` class.
+
+[source, c++]
+----
+  enum class memory_scope {
+  ...
+  sycl_ext_oneapi_experimental_cuda_cluster_group,
+  ...
+  };
+
+  inline constexpr auto sycl_ext_oneapi_experimental_cuda_memory_scope_cluster_group
+  = memory_scope::sycl_ext_oneapi_experimental_cuda_cluster_group;
+----
+
+
+[source,c++]
+----
+    class cluster_group {
+    public:
+    ...
+    static constexpr memory_scope fence_scope = 
+                memory_scope::sycl_ext_oneapi_experimental_cuda_cluster_fence;
+    }
+----
+
+To synchronize all workitems in the cluster group, `sycl::group_barrier` can be 
```

Comment:
```suggestion
To coordinate all work-items in the cluster group, `sycl::group_barrier` can be 
```

Reply:
Fixed in https://github.com/intel/llvm/pull/13594/commits/bf131725ad90260b2496948fc5b1d6ab0c80d372.